Add cps-iolib Ver2.0.0

Syunsuke-Okamoto opened the pull request.

Syunsuke-Okamoto committed:
8, Apr, 2019 : Test Implement cps-iolib
Bugfix Double access error for gpio63 and 69.

Syunsuke-Okamoto committed:
9, Apr, 2019 : Update cps-iolib Version from 1.0.5 to 1.0.6.
* Update Readme.md.
* Update Doxyfile.
* Fixed the cps-driver's Makefile.
* Without udelay cps_common_io.h

Syunsuke-Okamoto committed:
9, Apr, 2019 : Fixed bug list of Coverity Scan.
	I write [Result Flow Analize].

	* Version up cpsaio, cpscnt, 8250_cpscom, and cpsssi.

Syunsuke-Okamoto committed:
10,Apr,2019 : Fixed many Bugs.
(1) Fixed bug the rusult data in Coverity Scan.
* 8250_cpscom
* cpsaio
* cpscnt
* cpsssi

(2) Fixed spinlock cpsdio.
* cpsdio
* io (take in cps-iolib)
* cps-driver

(3) Update travis.yml
implement coverity-scan

ura1231 commented on cps-drivers/aio/cpsaio.c:
```diff
@@ -2353,7 +2355,7 @@ static long cpsaio_ioctl( struct file *filp, unsigned int cmd, unsigned long arg
 					valw = (unsigned short) d_ioc.val;
 					DEBUG_CPSAIO_COMMAND(KERN_INFO"DIRECT OUTPUT: [%lx]=%x\n",(unsigned long)( dev->baseAddr + d_ioc.addr ), valw );
-					cps_common_outw( (unsigned long)( dev->baseAddr + d_ioc.addr ) , valw );
+					contec_mcs341_outw( (unsigned long)( dev->baseAddr + d_ioc.addr ) , valw );
```
(About interrupt prohibition of lines 2355-2359)
Does it also have the purpose of preventing d_ioc.val from being updated by interrupts?
Interrupts are also disabled in contec_m cs341_outw.
For purposes of preventing unintended updates of cps_common_outw, I thought that lines 2355 and 2359 were unnecessary.

Syunsuke-Okamoto replied:
I was understanding.
I change from spin_lock_irqsave function to spin_lock function in IOCTL_CPSAIO_DIRECT_OUTPUT.(and IOCTL_CPSAIO_DIRECT_INPUT).

And , I change from spin_unlock_irqrestore function to spin_unlock function.

ura1231 commented on cps-drivers/aio/cpsaio.c:
```diff
@@ -2367,7 +2369,7 @@ static long cpsaio_ioctl( struct file *filp, unsigned int cmd, unsigned long arg
-					cps_common_inpw( (unsigned long)( dev->baseAddr + d_ioc.addr ) , &valw );
+					contec_mcs341_inpw( (unsigned long)( dev->baseAddr + d_ioc.addr ) , &valw );
 					DEBUG_CPSAIO_COMMAND(KERN_INFO"DIRECT INPUT:[%lx]=%x\n",(unsigned long)( dev->baseAddr + d_ioc.addr ), valw );
 					d_ioc.val = (unsigned long) valw;
```
If d_ioc.val has the possibility of being updated by an interrupt, I feel the need to call spin_unlock_irqrestore after copy_to_user () on line 2378.

Syunsuke-Okamoto replied:
I was understanding.
I change from spin_lock_irqsave function to spin_lock function in IOCTL_CPSAIO_DIRECT_OUTPUT.(and IOCTL_CPSAIO_DIRECT_INPUT).

And , I change from spin_unlock_irqrestore function to spin_unlock function.

Syunsuke-Okamoto replied:
You said, "Do you need to spinlock before copy_from_user (and copy_to_user)?"
I made new issue #12.
I am researching a new branch.

ura1231 reviewed (commented):
I listed two points of confirmation.Please confirm.

Syunsuke-Okamoto committed:
15, Apr, 2019 : Pullrequest commit.

Change from spin_lock_irqsave function to spin_lock function in
IOCTL_CPSAIO_DIRECT_OUTPUT.(and IOCTL_CPSAIO_DIRECT_INPUT).

And , change from spin_unlock_irqrestore function to spin_unlock
function.

Syunsuke-Okamoto committed:
15, Apr, 2019 : Bump up cps-iolib.doxyfile and test travis.yml

Syunsuke-Okamoto committed:
16, Apr, 2019 : travis.yml is implemented by coverity-scan. (#13)

Syunsuke-Okamoto committed:
16, Apr, 2019 : Update Readme.md

ura1231 reviewed (approved):
There is no problem.
Procees, please.
